File: web-sources/S8-fwks-palm/structs/columns/ColumnsStruct.js

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";
import { S8Object } from "/S8-api/S8Object.js";
import { Column } from "./Column.js";



S8WebFront.CSS_import('/S8-fwks-palm/structs/columns/Columns.css');


/**
 * 
 */
export class ColumnsStruct extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;


     /** @type {Column[]} */
    columns;


    /** */
    constructor() {
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("palm-struct-columns");
    }


    /** @returns */
    getEnvelope() {
        return this.wrapperNode;
    }


    /** @returns{S8Object[]} */
    getComponents() {
        return this.elements;
    }


    /** @param {Column[]} columns */
    S8_set_columns(columns) {
        this.columns = columns;
        S8WebFront.setS8Elements(this.wrapperNode, columns);
    }


    /** render */
    S8_render(){}


    /** dispose */
    S8_dispose(){}
}

File: sources/com/s8/pkgs/palm/components/structs/columns/ColumnsStruct.java
package com.s8.pkgs.palm.components.structs.columns;

import com.s8.api.web.S8WebFront;
import com.s8.pkgs.palm.components.PalmWebSources;
import com.s8.pkgs.ui.carbide.HTML_S8WebNode;


/**
 * 
 * @author pierreconvert
 *
 */
public class ColumnsStruct extends HTML_S8WebNode {
	
	
	
	public final static String WEBPATH = PalmWebSources.ROOT_WEBPATH + "/structs/columns/ColumnsStruct";
	
	

	
	/**
	 * 
	 * @param front
	 */
	
	public ColumnsStruct(S8WebFront front) {
		super(front, WEBPATH);
	}
	
	


	/**
	 * Columns
	 * @param size
	 */
	public void setColumns(Column... columns) {
		vertex.outbound().setObjectListField("columns", columns);
	}
	
	
}
